### DISCUSSION TITLE
Add an ability to mark a field as clean for elements

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#8879] [Ideas]
### TL/DR

Please add an additional function to mark fields as clean for elements

```PHP
$entry->markFieldAsClean('myCustomFieldName');
```

### Description

I often relate Elements with custom records instead of other elements for different reasons but eager loading custom records is not supported for Elements by default so I created custom logic for it.

When I have a bunch of elements that should be updated (eg via cron or bulk request) I would like to eager load all the records and set the elements field values. Currently I need to do this.

```php
foreach ($models as $entry){
    /** @var \craft\behaviors\ContentBehavior $contentBehavior */
    $contentBehavior = $entry->getBehavior('customFields');
    if($contentBehavior && is_numeric($contentBehavior->myCustomFieldName)){
        $id = (int)$contentBehavior->myCustomFieldName;
        $record = $myEagerLoadedRecords[$id]?? null;
        $contentBehavior->myCustomFieldName = $record;
    }
}
```  

Because if I did
```PHP
$entry->setFieldValue('myCustomFieldName', $record);
```
it would be marked as dirty and stored again, (in case eager loading is after the initialization) but I would like to prevent this.
So my feature request would be an additional function
```PHP
$entry->setFieldValue('myCustomFieldName', $record);
$entry->markFieldAsClean('myCustomFieldName');
```
```

**Comment by brandonkelly** [comment]
You can set your new field value like this instead, which will bypass it being marked as dirty:

```php
// ensure the value has been marked as normalized
// (you may be able to remove this if your normalizeValue() method can work with the eager-loaded data)
$entry->getFieldValue('myCustomFieldName');

// set the new value, bypassing setFieldValue()
$entry->getBehavior('customFields')->myCustomFieldName = $record;
```

Little hacky, but so is manually toggling a field’s clean/dirty state, so I’ll have to think about that one.
